File: src-backend/src/api/content/banner.js
import request from '@/utils/request'

// 查询标签列表
export function listBanner(query) {
  return request({
    url: '/api/admin/content/banner/page',
    method: 'get',
    params: query
  })
}

// 查询标签详细
export function getBanner(id) {
  return request({
    url: '/api/admin/content/banner/get',
    method: 'get',
    params: {
      id
    }
  })
}

// 新增标签配置
export function addBanner(data) {
  return request({
    url: '/api/admin/content/banner/insert',
    method: 'post',
    data: data
  })
}

// 修改标签配置
export function updateBanner(data) {
  return request({
    url: '/api/admin/content/banner/update',
    method: 'put',
    data: data
  })
}

// 删除标签配置
export function delBanner(id) {
  return request({
    url: '/api/admin/content/banner/delete',
    method: 'delete',
    params: {
      id
    }
  })
}

// 推荐状态修改
export function changeBannerStatus(id, status) {
  const data = {
    id,
    status
  }
  return request({
    url: '/api/admin/content/banner/status',
    method: 'put',
    data: data
  })
}

File: src-server/modules/content/routers/http/banner.go
// Copyright (c) 554949297@qq.com . 2022-2022 . All rights reserved

//go:build http
// +build http

package api

import (
	"github.com/zhouhp1295/g3"
	"github.com/zhouhp1295/g3-cms/boot"
	"github.com/zhouhp1295/g3-cms/modules/content/dao"
	"github.com/zhouhp1295/g3/net"
	"net/http"
)

type _contentBannerApi struct {
	net.BaseApi
}

var ContentBannerApi = &_contentBannerApi{
	net.BaseApi{Dao: dao.ContentBannerDao},
}

const (
	PermContentBannerList   = "content:banner:list"
	PermContentBannerQuery  = "content:banner:query"
	PermContentBannerAdd    = "content:banner:add"
	PermContentBannerEdit   = "content:banner:edit"
	PermContentBannerRemove = "content:banner:remove"
)

func init() {
	boot.RegisterAfterInstallFunction(func() {
		//基础接口
		g3.GetGin().Group("/api").
			Bind(http.MethodGet, "/admin/content/banner/page", ContentBannerApi.HandlePage, PermContentBannerQuery)
		g3.GetGin().Group("/api").
			Bind(http.MethodGet, "/admin/content/banner/get", ContentBannerApi.HandleGet, PermContentBannerQuery)
		g3.GetGin().Group("/api").
			Bind(http.MethodPost, "/admin/content/banner/insert", ContentBannerApi.HandleInsert, PermContentBannerAdd)
		g3.GetGin().Group("/api").
			Bind(http.MethodPut, "/admin/content/banner/update", ContentBannerApi.HandleUpdate, PermContentBannerEdit)
		g3.GetGin().Group("/api").
			Bind(http.MethodPut, "/admin/content/banner/status", ContentBannerApi.HandleUpdateStatus, PermContentBannerEdit)
		g3.GetGin().Group("/api").
			Bind(http.MethodDelete, "/admin/content/banner/delete", ContentBannerApi.HandleDelete, PermContentBannerRemove)
	})
}
